File: 205-isomorphic-strings/isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        int n=s.length();
        vector<int> sMap(256,0),tMap(256,0);

        for(int i=0;i<n;i++)
        {
            if((sMap[s[i]]!=0 || tMap[t[i]]!=0) &&
                (sMap[s[i]]!=t[i]+1 || tMap[t[i]]!=s[i]+1)) return false;
            sMap[s[i]]=t[i]+1;
            tMap[t[i]]=s[i]+1;
        }
        return true;
    }
};

File: 205-isomorphic-strings/isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        int n = s.length();

        int[] sMap = new int[256];
        int[] tMap = new int[256];

        for(int i=0;i<n;i++) {
            int sIndex = s.charAt(i);
            int tIndex = t.charAt(i);
            
            if((sMap[sIndex] > 0 && sMap[sIndex] != tIndex + 1) || 
                (tMap[tIndex] > 0 && tMap[tIndex] != sIndex + 1)) {
                    return false;
                }
            else {
                sMap[sIndex] = tIndex + 1;
                tMap[tIndex] = sIndex + 1;
            }
        }
        return true;
    }
}